### ISSUE TITLE
Importing Google Contacts VCF fails

### ISSUE THREAD

**Issue by sid3windr** [monicahq/monica#555]
Looks like there is an issue creating the birthday reminder:

```
(1/1) Exception

DateTime::__construct(): Failed to parse time string (Wish happy birthday to Firstname Lastname) at position 0 (W): The timezone could not be found in the database
in Carbon.php (line 291)
at DateTime->__construct('Wish happy birthday to Firstname Lastname', object(DateTimeZone))in Carbon.php (line 291)
at Carbon->__construct('Wish happy birthday to Firstname Lastname', null)in Carbon.php (line 324)
at Carbon::parse('Wish happy birthday to Firstname Lastname')in Reminder.php (line 94)
at Reminder::addBirthdayReminder(object(Contact), 'Wish happy birthday to Firstname Lastname', object(Carbon))in AddContactFromVCard.php (line 125)
at AddContactFromVCard->App\Jobs\{closure}(object(VCard), 2)in Collection.php (line 287)
at Collection->each(object(Closure))in AddContactFromVCard.php (line 138)
at AddContactFromVCard->handle()
at call_user_func_array(array(object(AddContactFromVCard), 'handle'), array())in BoundMethod.php (line 29)
at BoundMethod::Illuminate\Container\{closure}()in BoundMethod.php (line 87)
at BoundMethod::callBoundMethod(object(Application), array(object(AddContactFromVCard), 'handle'), object(Closure))in BoundMethod.php (line 31)
at BoundMethod::call(object(Application), array(object(AddContactFromVCard), 'handle'), array(), null)in Container.php (line 539)
at Container->call(array(object(AddContactFromVCard), 'handle'))in Dispatcher.php (line 94)
at Dispatcher->Illuminate\Bus\{closure}(object(AddContactFromVCard))in Pipeline.php (line 114)
at Pipeline->Illuminate\Pipeline\{closure}(object(AddContactFromVCard))in Pipeline.php (line 102)
at Pipeline->then(object(Closure))in Dispatcher.php (line 98)
at Dispatcher->dispatchNow(object(AddContactFromVCard), false)in CallQueuedHandler.php (line 42)
at CallQueuedHandler->call(object(SyncJob), array('commandName' => 'App\\Jobs\\AddContactFromVCard', 'command' => 'O:28:"App\\Jobs\\AddContactFromVCard":7:{s:12:"' . "\0" . '*' . "\0" . 'importJob";O:45:"Illuminate\\Contracts\\Database\\ModelIdentifier":2:{s:5:"class";s:13:"App\\ImportJob";s:2:"id";i:1;}s:19:"' . "\0" . '*' . "\0" . 'importedContacts";i:0;s:18:"' . "\0" . '*' . "\0" . 'skippedContacts";i:0;s:6:"' . "\0" . '*' . "\0" . 'job";N;s:10:"connection";N;s:5:"queue";N;s:5:"delay";N;}'))in Job.php (line 69)
at Job->fire()in SyncQueue.php (line 42)
at SyncQueue->push(object(AddContactFromVCard))in Dispatcher.php (line 184)
at Dispatcher->pushCommandToQueue(object(SyncQueue), object(AddContactFromVCard))in Dispatcher.php (line 159)
at Dispatcher->dispatchToQueue(object(AddContactFromVCard))in Dispatcher.php (line 73)
at Dispatcher->dispatch(object(AddContactFromVCard))in helpers.php (line 378)
at dispatch(object(AddContactFromVCard))in SettingsController.php (line 181)
at SettingsController->storeImport(object(ImportsRequest))
at call_user_func_array(array(object(SettingsController), 'storeImport'), array(object(ImportsRequest)))in Controller.php (line 55)
at Controller->callAction('storeImport', array(object(ImportsRequest)))in ControllerDispatcher.php (line 44)
at ControllerDispatcher->dispatch(object(Route), object(SettingsController), 'storeImport')in Route.php (line 203)
at Route->runController()in Route.php (line 160)
at Route->run()in Router.php (line 574)
at Router->Illuminate\Routing\{closure}(object(Request))in Pipeline.php (line 30)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in Authenticate.php (line 28)
at Authenticate->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in CheckVersion.php (line 32)
at CheckVersion->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in CheckLocale.php (line 25)
at CheckLocale->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in SubstituteBindings.php (line 41)
at SubstituteBindings->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in VerifyCsrfToken.php (line 65)
at VerifyCsrfToken->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in ShareErrorsFromSession.php (line 49)
at ShareErrorsFromSession->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in StartSession.php (line 64)
at StartSession->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in AddQueuedCookiesToResponse.php (line 37)
at AddQueuedCookiesToResponse->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in EncryptCookies.php (line 59)
at EncryptCookies->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in Pipeline.php (line 102)
at Pipeline->then(object(Closure))in Router.php (line 576)
at Router->runRouteWithinStack(object(Route), object(Request))in Router.php (line 535)
at Router->dispatchToRoute(object(Request))in Router.php (line 513)
at Router->dispatch(object(Request))in Kernel.php (line 176)
at Kernel->Illuminate\Foundation\Http\{closure}(object(Request))in Pipeline.php (line 30)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in Debugbar.php (line 51)
at Debugbar->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in CheckForMaintenanceMode.php (line 46)
at CheckForMaintenanceMode->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in Pipeline.php (line 102)
at Pipeline->then(object(Closure))in Kernel.php (line 151)
at Kernel->sendRequestThroughRouter(object(Request))in Kernel.php (line 116)
at Kernel->handle(object(Request))in index.php (line 52)`
```

Also, I'm fairly sure this "Wish happy birthday to" reminder is no longer needed as new reminders are auto-created for every birthday as well, are they not? (titled "Birthday of Firstname")

**Comment by scott-joe** [comment]
Having the same issues:

```log
(1/1) ExceptionDateTime::__construct(): Failed to parse time string (Wish happy birthday to Jeff Williams) at position 0 (W): The timezone could not be found in the database
--
in Carbon.php (line 291)
at DateTime->__construct('Wish happy birthday to Jeff Williams', object(DateTimeZone))in Carbon.php (line 291)
at Carbon->__construct('Wish happy birthday to Jeff Williams', null)in Carbon.php (line 324)
at Carbon::parse('Wish happy birthday to Jeff Williams')in Reminder.php (line 94)
at Reminder::addBirthdayReminder(object(Contact), 'Wish happy birthday to Jeff Williams', object(Carbon))in AddContactFromVCard.php (line 125)
at AddContactFromVCard->App\Jobs\{closure}(object(VCard), 13)in Collection.php (line 287)
at Collection->each(object(Closure))in AddContactFromVCard.php (line 138)
at AddContactFromVCard->handle()
at call_user_func_array(array(object(AddContactFromVCard), 'handle'), array())in BoundMethod.php (line 29)
at BoundMethod::Illuminate\Container\{closure}()in BoundMethod.php (line 87)
at BoundMethod::callBoundMethod(object(Application), array(object(AddContactFromVCard), 'handle'), object(Closure))in BoundMethod.php (line 31)
at BoundMethod::call(object(Application), array(object(AddContactFromVCard), 'handle'), array(), null)in Container.php (line 539)
at Container->call(array(object(AddContactFromVCard), 'handle'))in Dispatcher.php (line 94)
at Dispatcher->Illuminate\Bus\{closure}(object(AddContactFromVCard))in Pipeline.php (line 114)
at Pipeline->Illuminate\Pipeline\{closure}(object(AddContactFromVCard))in Pipeline.php (line 102)
at Pipeline->then(object(Closure))in Dispatcher.php (line 98)
at Dispatcher->dispatchNow(object(AddContactFromVCard), false)in CallQueuedHandler.php (line 42)
at CallQueuedHandler->call(object(SyncJob), array('commandName' => 'App\\Jobs\\AddContactFromVCard', 'command' => 'O:28:"App\\Jobs\\AddContactFromVCard":7:{s:12:"' . "\0" . '*' . "\0" . 'importJob";O:45:"Illuminate\\Contracts\\Database\\ModelIdentifier":2:{s:5:"class";s:13:"App\\ImportJob";s:2:"id";i:11;}s:19:"' . "\0" . '*' . "\0" . 'importedContacts";i:0;s:18:"' . "\0" . '*' . "\0" . 'skippedContacts";i:0;s:6:"' . "\0" . '*' . "\0" . 'job";N;s:10:"connection";N;s:5:"queue";N;s:5:"delay";N;}'))in Job.php (line 69)
at Job->fire()in SyncQueue.php (line 42)
at SyncQueue->push(object(AddContactFromVCard))in Dispatcher.php (line 184)
at Dispatcher->pushCommandToQueue(object(SyncQueue), object(AddContactFromVCard))in Dispatcher.php (line 159)
at Dispatcher->dispatchToQueue(object(AddContactFromVCard))in Dispatcher.php (line 73)
at Dispatcher->dispatch(object(AddContactFromVCard))in helpers.php (line 378)
at dispatch(object(AddContactFromVCard))in SettingsController.php (line 181)
at SettingsController->storeImport(object(ImportsRequest))
at call_user_func_array(array(object(SettingsController), 'storeImport'), array(object(ImportsRequest)))in Controller.php (line 55)
at Controller->callAction('storeImport', array(object(ImportsRequest)))in ControllerDispatcher.php (line 44)
at ControllerDispatcher->dispatch(object(Route), object(SettingsController), 'storeImport')in Route.php (line 203)
at Route->runController()in Route.php (line 160)
at Route->run()in Router.php (line 574)
at Router->Illuminate\Routing\{closure}(object(Request))in Pipeline.php (line 30)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in Authenticate.php (line 28)
at Authenticate->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in CheckVersion.php (line 32)
at CheckVersion->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in CheckLocale.php (line 25)
at CheckLocale->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in SubstituteBindings.php (line 41)
at SubstituteBindings->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in VerifyCsrfToken.php (line 65)
at VerifyCsrfToken->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in ShareErrorsFromSession.php (line 49)
at ShareErrorsFromSession->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in StartSession.php (line 64)
at StartSession->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in AddQueuedCookiesToResponse.php (line 37)
at AddQueuedCookiesToResponse->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in EncryptCookies.php (line 59)
at EncryptCookies->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in Pipeline.php (line 102)
at Pipeline->then(object(Closure))in Router.php (line 576)
at Router->runRouteWithinStack(object(Route), object(Request))in Router.php (line 535)
at Router->dispatchToRoute(object(Request))in Router.php (line 513)
at Router->dispatch(object(Request))in Kernel.php (line 176)
at Kernel->Illuminate\Foundation\Http\{closure}(object(Request))in Pipeline.php (line 30)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in Debugbar.php (line 51)
at Debugbar->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in CheckForMaintenanceMode.php (line 46)
at CheckForMaintenanceMode->handle(object(Request), object(Closure))in Pipeline.php (line 148)
at Pipeline->Illuminate\Pipeline\{closure}(object(Request))in Pipeline.php (line 53)
at Pipeline->Illuminate\Routing\{closure}(object(Request))in Pipeline.php (line 102)
at Pipeline->then(object(Closure))in Kernel.php (line 151)
at Kernel->sendRequestThroughRouter(object(Request))in Kernel.php (line 116)
at Kernel->handle(object(Request))in index.php (line 52)
```

**Comment by djaiss** [comment]
Comment from user:

> I installed on Heroku and enabled debug mode. Turns out someone just created an issue for this. It has to do with missing timezones in the timezones table, but it's not the table's fault bc the field doesn't have a timezone on import... I'll add my log to theirs

**Comment by scott-joe** [comment]
So, your unit tests were right, there was just a piece of code that I assume was doing it "the old way." According to the unit test and this bit of code, this seems to be the proper way to add a birthday reminder.
<img width="447" alt="screenshot 2017-08-22 18 13 37" src="https://user-images.githubusercontent.com/2026866/29590066-19e9f5b2-8766-11e7-8dd3-4ae00224b4e1.png">

This is what it was doing. I commented out the 'broken bits'
<img width="523" alt="screenshot 2017-08-22 18 13 13" src="https://user-images.githubusercontent.com/2026866/29590067-19fa5e16-8766-11e7-9fea-31eb2fab9710.png">

I added my contacts and it went fine. I'm going to test it out with my full 400 contacts instead of the 20 test file to see what happens.
